### ISSUE TITLE
Keine Daten in der TTN Console, war: Clonen des Github repos geht nicht

### ISSUE THREAD

**Issue by SchneHa** [verschwoerhaus/ttn-ulm-doku#2]
Das Clonen des Github repos schlägt fehl:

Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by dermatthias** [comment]
Vermutlich versuchst du via SSH zu clonen (publickey). Hast du deine SSH Keys entsprechend in deinem Github Profil hinterlegt? Dann klappt das. Sonst geht clonen via HTTPS ohne weitere Anforderungen.

(Das ganze hat auch nichts mit diesem Repo zu tun, eher mit Github an sich)

**Comment by SchneHa** [comment]
Ja, ich habe versucht, über ssh zu clonen, in allen anderen Fällen hat dies auch immer funktioniert. Bei meiner Suche nach dem Fehler habe ich herausgefunden, dass in der entsprechenden Zeile in deinem Tutorial ein Fehler ist, wenn man statt

git clone git@github.com:verschwoerhaus/ttn-ulm-node-dragino.git geosender

schreibt:

git clone https://github.com/verschwoerhaus/ttn-ulm-node-dragino geosender

funktioniert es.

Ich habe mich dann nach deiner Anleitung gerichtet und auch auf dem Dragino Hat die im Dragino Wiki beschriebenen Änderungen bezüglich GPS erfolgreich durchgeführt. Trotzdem sehe ich in meiner TTN Konsole keine GPS Daten. Es kommen zwar Daten rein, aber die Payload ist immer 00 00 00 00 00 00. Ein 

Ich denke es liegt daran, dass der gpslogger nicht richtig funktioniert. Im Gegensatz zum geotagger erhalte ich hier immer eine Fehlermeldung, die wie hier aussieht:

pi@RPiEIGHT:~/geosender/ttnsender/main $ sudo systemctl status gpslogger.service
● gpslogger.service - TTN GPS Logger
   Loaded: loaded (/lib/systemd/system/gpslogger.service; enabled)
   Active: failed (Result: exit-code) since Di 2018-02-27 15:26:13 CET; 4min 8s ago
  Process: 2004 ExecStart=/home/pi/geosender/gpslogger/gpsd_start.sh (code=exited, status=1/FAILURE)
 Main PID: 2004 (code=exited, status=1/FAILURE)

Feb 27 15:26:08 RPiEIGHT sudo[2005]: root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/usr/sbin/gpsd /dev/ttyS0 -F /var/run/gpsd.sock
Feb 27 15:26:08 RPiEIGHT sudo[2005]: pam_unix(sudo:session): session opened for user root by (uid=0)
Feb 27 15:26:08 RPiEIGHT sudo[2005]: pam_unix(sudo:session): session closed for user root
Feb 27 15:26:08 RPiEIGHT gpsd[2013]: gpsd:ERROR: can't bind to IPv4 port gpsd, Address already in use
Feb 27 15:26:13 RPiEIGHT gpsd_start.sh[2004]: Traceback (most recent call last):
Feb 27 15:26:13 RPiEIGHT gpsd_start.sh[2004]: File "/home/pi/geosender/gpslogger/gps.py", line 2, in <module>
Feb 27 15:26:13 RPiEIGHT gpsd_start.sh[2004]: from gps3 import gps3
Feb 27 15:26:13 RPiEIGHT gpsd_start.sh[2004]: ImportError: No module named gps3
Feb 27 15:26:13 RPiEIGHT systemd[1]: gpslogger.service: main process exited, code=exited, status=1/FAILURE
Feb 27 15:26:13 RPiEIGHT systemd[1]: Unit gpslogger.service entered failed state.

Was geht hier schief oder besser: Was kann ich ändern, damit es funktioniert?

**Comment by stkdiretto** [comment]
Ja, genau -- wenn du mit git@github.com:organisation/repo clonen willst, muss der SSH-Key eingetragen sein. [Siehe auch hier](https://stackoverflow.com/questions/16245606/could-not-read-from-remote-repository). @dermatthias fuer EinsteigerInnen ist die Variante ueber https vielleicht barriereaermer, auch wenn sie spaeter vielleicht mal mit publickey arbeiten wollten/sollten, was meinst du?

@SchneHa welche python-version hast du denn installiert? Ist `python-gps` auch installiert? Was sagt `dpkg -L python-gps` auf der Kommandozeile?

**Comment by stkdiretto** [comment]
~~dermatthias ich kann den build hier grad auch nicht reproduzieren – per default installiert die Vorgehensweise mir ein 2er-Python, da gibts kein pip3 und kein gps3.~~

ich nehme alles zurueck. @SchneHa versuch mal `sudo pip install gps3` und probier's dann nochmal.

**Comment by SchneHa** [comment]
Python 2.7 und python-gps scheint installiert zu sein.

pi@RPiEIGHT:~ $ dpkg -L python-gps
/.
/usr
/usr/lib
/usr/lib/python2.7
/usr/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages/gps
/usr/lib/python2.7/dist-packages/gps/__init__.py
/usr/lib/python2.7/dist-packages/gps/client.py
/usr/lib/python2.7/dist-packages/gps/fake.py
/usr/lib/python2.7/dist-packages/gps/gps.py
/usr/lib/python2.7/dist-packages/gps/misc.py
/usr/lib/python2.7/dist-packages/gps/clienthelpers.so
/usr/lib/python2.7/dist-packages/gps/packet.so
/usr/lib/python2.7/dist-packages/gps-3.11.egg-info
/usr/share
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/gpsfake.1.gz
/usr/share/man/man1/gpscat.1.gz
/usr/share/man/man1/gpsprof.1.gz
/usr/share/doc
/usr/share/doc/python-gps
/usr/share/doc/python-gps/copyright
/usr/share/doc/python-gps/changelog.Debian.gz
/usr/share/doc/python-gps/changelog.gz
/usr/bin
/usr/bin/gpscat
/usr/bin/gpsfake
/usr/bin/gpsprof

**Comment by SchneHa** [comment]
… aber python3 ist auch installiert:

pi@RPiEIGHT:~ $ sudo apt-get install python3
Paketlisten werden gelesen... Fertig
Abhängigkeitsbaum wird aufgebaut.       
Statusinformationen werden eingelesen.... Fertig
python3 ist schon die neueste Version.
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.

**Comment by SchneHa** [comment]
python2.6, python2.7, python3 und python3.4 finde ich in /usr/lib

**Comment by SchneHa** [comment]
einzige Fehlermeldung, die ich während der Installation erhalten habe:

pi@RPiEIGHT:~ $ sudo apt-get install python-gps gcc-6 g++-6
Paketlisten werden gelesen... Fertig
Abhängigkeitsbaum wird aufgebaut.       
Statusinformationen werden eingelesen.... Fertig
Note, selecting 'libopencascade-modeling-6.5.0' for regex 'g++-6'
E: Paket gcc-6 kann nicht gefunden werden.

**Comment by stkdiretto** [comment]
@SchneHa top, danke – hast du den Tipp in https://github.com/verschwoerhaus/ttn-ulm-doku/issues/2#issuecomment-368927014 ausprobiert? `sudo pip install gps3`

**Comment by SchneHa** [comment]
Danke für den Tipp, habe ich jetzt gemacht. Scheint zu laufen, aber Daten kommen in der TTN Konsole immer noch nicht an.

**Comment by SchneHa** [comment]
So, es läuft jetzt schon eine ganze Zeit aber Daten kommen immer noch nicht an. Payload ist immer noch 0. Irgendwo scheint noch ein Fehler zu liegen. Auch, wenn ich im Terminal cgps eingebe, werden keine Daten angezeigt.

**Comment by SchneHa** [comment]
cgps zeigt jetzt Daten an, aber auf TTN kommt nichts davon an.

**Comment by SchneHa** [comment]
Das Problem scheint der geotagger zu sein, sudo journalctl -xe zeigt reihenweise Zeilen wie diese an:

Mär 01 18:05:24 RPiEIGHT geotagger[747]: 17:58:28: EV_TXCOMPLETE (includes waiting for RX windows)
Mär 01 18:05:24 RPiEIGHT geotagger[747]: Failed!Sending: 0.000000,0.000000
Mär 01 18:05:24 RPiEIGHT geotagger[747]: Packet queued

Muss mich korrigieren: Die /home/pi/geo/geo.dat scheint erst gar nicht angelegt zu werden. Also scheint der data_stream nicht gefunden zu werden. Hat das Problem also doch mit gps3 zu tun? 
In der gps.py finde ich die Zeilen "gps_socket = gps3.GPSDSocket()" und "data_stream = gps3.DataStream()", liegt hier der Hund begraben?

**Comment by stkdiretto** [comment]
Ich versuch morgen mal das Problem mit der Hardware zu rekonstruieren.

**Comment by stkdiretto** [comment]
Ich bekomm gar nicht erst Daten aus dem GPS raus (RasPi3, Raspbian stretch). Sorry, ich bleib mal weiter dran.

**Comment by SchneHa** [comment]
Manchmal kommen bei mir auch keine und ich habe noch nicht raus bekommen, woran das liegt (RasPi3, Rapbian Jessie). Manchmal hilft ein Neustart.

**Comment by SchneHa** [comment]
Momentan geht auch bei mir nix mehr, keine Daten aus dem GPS.

**Comment by SchneHa** [comment]
So, bei mir geht es jetzt!!!

Was ich gemacht habe:

* Auf dem Dragino Hat DI01 mit WiringPi pin 4 (16; BCM23) und DI02 mit WiringPi pin 5 (18; BCM24) verbunden. Weil ich die Version 1.0 des Boards habe und in der aktuellen Version diese Pins verbunden sind.

* WiringPi installiert

(Ich weiß nicht, ob die ersten beiden Schritte nötig waren, aber so funktioniert es)

* In der /home/pi/geosender/gpslogger/gps.py die Zeile 7 abgeändert, 
alt:
f = open('/home/pi/geo/geo.dat', 'w')
neu: 
f = open('/home/pi/geosender/data/geo.dat', 'w') 
und ebenso die Zeile 8, 
alt:
fa = open('/home/pi/geo/geo_all.dat', 'a')
neu:
fa = open('/home/pi/geosender/data/geo_all.dat', 'a')

* Die Berechtigung der beiden Dateien /home/pi/geosender/data/geo_all.dat und /home/pi/geosender/data/geo.dat von 644 auf 666 geändert (weil gps.py mit der Meldung ausgestiegen war, dass die Berechtigung falsch sei)

* In der Datei /home/pi/geosender/ttnsender/main/geotagger.cpp die Zeile 134 geändert, 
alt:
ifstream inf("/../../geosender/data/geo.dat");
neu:
ifstream inf("/home/pi/geosender/data/geo.dat");

* mit sudo make neu compiliert, that's it!

**Comment by SchneHa** [comment]
Ein Problem gibt es noch: Die gps.py wird nicht automatisch gestartet. Eingeschaltet ist es mit sudo systemctl enable gpslogger.service, /home/pi/geosender/gpslogger/gpsd_start.sh ist vorhanden und hat den richtigen Inhalt, aber es wird nach einem reboot nichts in die geo_all.dat geschrieben und daher kommen nur alte Werte bei TTN an.
